torchgeo/trainers/segmentation.py

```diff
@@ -65,9 +65,12 @@ def config_task(self) -> None:
         if self.hyperparams["loss"] == "ce":
             ignore_value = -1000 if self.ignore_index is None else self.ignore_index
 
-            class_weights = (
-                torch.FloatTensor(self.class_weights) if self.class_weights else None
-            )
+            class_weights = None
+            if isinstance(self.class_weights, torch.Tensor):
+                class_weights = self.class_weights.to(dtype=torch.float32)
+            elif hasattr(self.class_weights, "__array__") or self.class_weights:
+                class_weights = torch.tensor(self.class_weights, dtype=torch.float32)
```

Comment:
We also want to be able to support lists and other sequences. How about:
```suggestion
            if self.class_weights is not None:
                class_weights = torch.tensor(self.class_weights, dtype=torch.float32)
```

Reply:
Yes, other sequence types (including lists) are already covered by the `or self.class_weights` part of the branch test. This mimics the previous behaviour of the function (that only supported sequences), where `class_weights` would be `None` if the input sequence was non-null but empty.

It became apparent during development that we needed a separate branch to handle PyTorch tensors, as copying a tensor via the `torch.tensor` function raises warnings as follows:

```
>>> import torch
>>> a = torch.arange(10)
>>> b = torch.tensor(a)
<stdin>:1: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
```